core/objects.c

```diff
@@ -189,35 +189,67 @@ coap_status_t object_write(lwm2m_context_t * contextP,
                            char * buffer,
                            int length)
 {
-    coap_status_t result;
+    coap_status_t result = NO_ERROR;
     lwm2m_object_t * targetP;
     lwm2m_tlv_t * tlvP = NULL;
     int size = 0;
 
     targetP = prv_find_object(contextP, uriP->objectId);
-    if (NULL == targetP) return NOT_FOUND_4_04;
-    if (NULL == targetP->writeFunc) return METHOD_NOT_ALLOWED_4_05;
-
-    if (LWM2M_URI_IS_SET_RESOURCE(uriP))
-    {
-        size = 1;
-        tlvP = lwm2m_tlv_new(size);
-        if (tlvP == NULL) return COAP_500_INTERNAL_SERVER_ERROR;
-
-        tlvP->flags = LWM2M_TLV_FLAG_TEXT_FORMAT | LWM2M_TLV_FLAG_STATIC_DATA;
-        tlvP->type = LWM2M_TYPE_RESSOURCE;
-        tlvP->id = uriP->resourceId;
-        tlvP->length = length;
-        tlvP->value = (uint8_t*)buffer;
+    if (NULL == targetP) {
```

Comment:
I meant the code insid the if (NULL == targetP) : the object_create() will fail each time.
You should change to:
```C
if (NULL == targetP)
{
    bootstrap_failed(contextP);
    return COAP_404_NOT_FOUND;
}
```